modules/channel-web/src/backend/api.ts

```diff
@@ -305,7 +305,14 @@ export default async (bp: typeof sdk, db: Database) => {
       const conversation = await req.messaging.conversations.get(conversationId)
       const messages = await req.messaging.messages.list(conversationId, config.maxMessagesHistory)
 
-      const filteredMessages = messages.filter(m => m.payload.type !== 'visit')
+      const filteredMessages = messages.filter(m => {
```

Comment:
```ts
// this function scope can (and probably will) expand to other types as well with a switch case on the type
// we do something similar in the cms to determine weather there are translated fields or not
const notEmptyPayload = (payload) => payload.type === 'text' ? !!payload.text : true

const displayableMessages = messages.fiter( ({payload}) => payload.type !== 'visit' && noEmptyPayload(payload) )
```

Reply:
I'm a bit worried though, why are there empty messages in the 1st place ?

Reply:
I think it's from messenger...
I'll try check the whole flow to see where that happens